src/python/zquantum/qaoa/maxcut.py

```diff
@@ -2,37 +2,103 @@
 import numpy as np
 from itertools import combinations
 from random import uniform
-
-
 from openfermion import QubitOperator
 from zquantum.core.utils import dec2bin
-from zquantum.core.graph import generate_graph_node_dict
+from zquantum.core.graph import generate_graph_node_dict, generate_graph_from_specs
+from typing import Dict, List, Union
+
+from .farhi_ansatz import QAOAFarhiAnsatz
+
+
+def create_farhi_qaoa_circuits(
```

Comment:
I think this function could be `create_circuits_for_hamiltonians`, and it could take `ansatz` as parameter – then we would have more flexibility if someone wants to use some other ansatz.
On the other hand we don't have any other ansatzes at the moment, so this might be premature optimization.
WDYT: @simonwa7 ?

Reply:
It has not been changed as it is not clear yet how much it can be generalized (depends on the detail of potential new ansatzs)

Reply:
Ok 👍